bazel/cc_toolchains/clang_configuration.bzl

```
@@ -18,25 +18,34 @@ def _run(repository_ctx, cmd):
     return exec_result
 
 def _clang_version(version_output):
-    """Returns clang's major version number, or None if not found."""
+    """Returns version information, or a (None, "unknown") tuple if not found.
+
+    Returns both the major version number (14) and the full version number for
+    caching.
+    """
+    clang_version = None
+    clang_version_for_cache = "unknown"
+
     version_prefix = "clang version "
     version_start = version_output.find(version_prefix)
     if version_start == -1:
         # No version
-        return None
+        return (clang_version, clang_version_for_cache)
     version_start += len(version_prefix)
 
-    # Find a dot to indicate something like 'clang version 14.0.6'.
-    version_dot = version_output.find(".", version_start)
-    if version_dot == -1:
-        return None
+    # Find the newline.
+    version_newline = version_output.find("\n", version_start)
+    if version_newline == -1:
+        return (clang_version, clang_version_for_cache)
+    clang_version_for_cache = version_output[version_start:version_newline]
 
-    # Make sure the dot was on the same line as the version.
-    if version_output.find("\n", version_start) < version_dot:
-        return None
+    # Find a dot to indicate something like 'clang version 14.0.6', and grab the
+    # major version.
+    version_dot = version_output.find(".", version_start)
+    if version_dot != -1 and version_dot < version_newline:
+        clang_version = int(version_output[version_start:version_dot])
 
-    # Return the version as int.
-    return int(version_output[version_start:version_dot])
+    return (clang_version, clang_version_for_cache)
```

```suggestion
    return (clang_version, clang_version_for_cache.translate({ord('"'): '_', ord('\\'): '_'})
```
May as well escape this here.